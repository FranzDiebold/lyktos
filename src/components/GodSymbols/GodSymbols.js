import React from 'react';
import PropTypes from 'prop-types';

import { Tag } from 'bloomer';

function GodSymbols(props) {
    const { symbols } = props;

    const symbolTags = symbols.map(symbol => <Tag key={symbol}>{symbol}</Tag>);

    return (
        <div>
            <span className="has-text-grey is-italic">Symbols</span>
            <div className="tags are-medium">
                {symbolTags}
            </div>
        </div>
    );
}

GodSymbols.propTypes = {
    symbols: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default GodSymbols;
