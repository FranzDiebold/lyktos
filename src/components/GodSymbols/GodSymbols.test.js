import React from 'react';
import renderer from 'react-test-renderer';

import GodSymbols from './GodSymbols';

it('renders god symbols correctly', () => {
    const tree = renderer
        .create(<GodSymbols symbols={['Symbol 1', 'Symbol 2']} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
