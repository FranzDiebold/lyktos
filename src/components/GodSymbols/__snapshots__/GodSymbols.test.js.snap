// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders god symbols correctly 1`] = `
<div
  className="tags are-medium"
>
  <span
    className="tag"
  >
    Symbol 1
  </span>
  <span
    className="tag"
  >
    Symbol 2
  </span>
</div>
`;
